src/main/kotlin/racing/domain/RaceResult.kt

```diff
@@ -0,0 +1,11 @@
+package racing.domain
+
+data class RaceResult(
+    val results: List<Result>,
+) {
+    fun getWinnerNames(): List<String> {
+        val lastResult = results.maxByOrNull(Result::round) ?: return emptyList()
+
+        return lastResult.getWinnerNames()
+    }
+}
```

Comment:
```suggestion
data class RaceResult(
    val results: List<Result>,
) {
    private val lastResult = results.maxBy { it.round }
    fun getWinnerNames(): List<String> {
        return lastResult.getWinnerNames()
    }
}
```

지금의 RaceResult를 저는 위와 같이 표현하고 싶어요. 위와 같이 표현한 이유는 

- RaceResult의 results는 로직 상 빈 리스트로 들어와서는 안된다.
- getWinnerNames에는 항상 results의 마지막 라운드에 대한 정보가 들어있다.

이러한 이유로, maxBy { } 를 통하여 제공되는 results가 없다면 emptyList로 반환하는 것보다, `NoSuchElementException` 예외를 발생시키도록 하였습니다. 또한 results는 data class이고, 불변으로 동작을 하기에 처음에 들어온 results를 이용하여 바로 lastResult를 초기화 했습니다.  😊

Result class의 getWinnerNames도 위와 같이 표현이 가능해 보이는데요. 생각해 보시고 피드백 반영 고민해 보세요.

Reply:
오 maxByOrNull을 쓴 것은 maxBy가 deprecated 되었더라구요
그래서 null 처리를 해야해서 그냥 emptyList()를 반환하게 했는데 말씀하신대로 경주가 시작되면 결과는 항상 있기 때문에
예외가 발생하는 게 더 적절한 것 같습니다 👍🏻 

피드백주신대로 수정해봤는데 제 느낀 점으로는 더 코틀린스러워진 느낌이에요 😄 
그리고 게터를 이렇게 사용할 수가 있는 것도 신기합니다

Reply:
해당 코드를 인텔리제이에서 Tool > Kotlin >  Show Kotlin Byte Code를 사용하여 어떻게 lastResult를 만들어지는지도 한번 확인해보시겠어요?

예상과 다른 동작을 할거에요!

Reply:
오 객체 생성자가 호출될 때 생성자 내부에서 초기화하네요
초기화 선언을 같이해서 이렇게 동작하는 것 같아요

Reply:
맞아요! 자바를 배우고, 코틀린을 공부할 때 코틀린 코드가 어떻게 바이트 코드로 변환되는지, 자바로 디컴파일하면 어떻게 되는지 확인하면서 공부하면 코드 동작에 대해서 더 깊게 이해할 수가 있어요!